sov-modules/sov-modules-macros/src/dispatch/common.rs

```diff
@@ -143,3 +142,26 @@ pub(crate) fn parse_generic_params(generics: &Generics) -> Result<Ident, syn::Er
 
     Ok(generic_param.clone())
 }
+
+pub fn get_attribute_values(item: &syn::DeriveInput, attribute_name: &str) -> Vec<String> {
+    let mut values = vec![];
+
+    // Find the attribute with the given name on the root item
+    if let Some(attr) = item.attrs.iter().find(|attr| attr.path.is_ident(attribute_name)) {
+        if let Ok(meta) = attr.parse_meta() {
+            if let Meta::List(list) = meta {
+                values.extend(
+                    list.nested
+                        .iter()
+                        .filter_map(|nested| match nested {
+                            NestedMeta::Lit(Lit::Str(lit_str)) => Some(lit_str.value()),
+                            _ => None,
+                        })
+                        .collect::<Vec<_>>(),
+                );
+            }
+        }
+    }
+
```

Comment:
We will always want to specify the serialization via macro attribute, so if
`#[serialization(..)]` is missing. We should emit a nice compile time error. 
[Here](https://github.com/Sovereign-Labs/sovereign/blob/research/sov-modules/sov-modules-macros/src/module_info.rs#L143) is an example of how to emit an error from macro and a corresponding test is in [derive_on_enum_not_supported.rs](https://github.com/Sovereign-Labs/sovereign/blob/research/sov-modules/sov-modules-macros/tests/module_info/derive_on_enum_not_supported.rs) and [derive_on_enum_not_supported.stderr](https://github.com/Sovereign-Labs/sovereign/blob/research/sov-modules/sov-modules-macros/tests/module_info/derive_on_enum_not_supported.stderr).